File: facade-pattern/java/src/FileModule.java
package src;

// 文件模块
public class FileModule {

   public void readFile() {
      System.out.println(this.getClass().getName() + "::readFile() 读取文件。");
   }

   public void saveFile() {
      System.out.println(this.getClass().getName() + "::saveFile() 保存文件。");
   }
}

File: facade-pattern/go/src/FileModule.go
package src

import "fmt"

// 文件模块
type FileModule struct {
}

func (f *FileModule) ReadFile() {
  fmt.Println("FileModule::ReadFile() 读取文件。")
}

func (f *FileModule) SaveFile() {
  fmt.Println("FileModule::SaveFile() 保存文件。")
}
